### ISSUE TITLE
no-shadow does not catch shadowing when the shadowing is inside function properties of object literals

### ISSUE THREAD

**Issue by gcochard** [eslint/eslint#1322]
The following pattern is caught:

``` javascript
function outer(x){
    function second(){
        var x = 'fdsa';
        return x;
    }
    return second();
}
console.log(outer('x'));
```

But this is not caught:

``` javascript
function outer(x){
    'use strict';
    console.log('in outer, x: %j',x);
    var b = { 
        first: function(){
            console.log('in first, x: %j',x);
        },  
        second: function(){
            var x = 'fdsa';
            return x;
        }   
    };  
    b.first();
    return b.second();
}
console.log('result of second: %j',outer('b'));
```

These are both equivalent examples of shadowing and should thus both be flagged by the rule.
